### PR TITLE
 Handle API_TARGET_BLOCKED and RESOURCE_EXHAUSTED

### LINKED ISSUE #573: "Request blocked due to unsupported error code: 122"
Hello ESP devs,

We've been having some issues using ESP with Cloud Functions, specifically, we think, the Service Control API. For a subset of our traffic, and only on a few API keys at a time (leading us to believe it's related to quota control somehow), we get the following JSON response served as a `500 Internal Server Error`:

```json
{
  "code": 13,
  "message": "Request blocked due to unsupported error code: 122",
  "details": [
    {
      "@type": "type.googleapis.com/google.rpc.DebugInfo",
      "stackEntries": [],
      "detail": "service_control"
    }
  ]
}
```

The line of code that produces this error is from ESP, not Nginx:
https://github.com/cloudendpoints/esp/blob/9c50be735adf66180427a9044fffa023a4b06405/src/api_manager/service_control/proto.cc#L1346

So, we're wondering what the error code is, and how we might go about fixing it? Thank you in advance for your help

### PR DESCRIPTION
**Pull request by kh-chang**
Fixes #573

### CODE REVIEW

File: repositories.bzl [hunk 1]
```diff
@@ -572,14 +574,25 @@ cc_proto_library(
...
+    #
+    # Instructions for updating googleapis:
+    #  1) fork from https://github.com/googleapis/googleapis.git
+    #  2) make a branch
+    #  3) remove BUILD files `$find . -type f -name '*BUILD*' | xargs rm`
+    #  4) commit and push
+    #  5) Update `commit` and `remote` in `new_git_repository`
+    #  6) Update the below comment to reflect the new version
+    #
+    # Corresponds to 9a02c5acecb43f38fae4fa52c6420f21c335b888 of googleapis/googleapis.git
```
Comment (qiwzhang) [inline repositories.bzl]: also specify the date for this SHA

File: src/nginx/t/check_api_target_blocked.t [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+################################################################################
+
+my $response = ApiManager::http_get($NginxPort,'/shelves?key=this-is-an-api-key');
+
+is($t->waitforfile("$t->{_testdir}/${report_done}"), 1, 'Report body file ready.');
+$t->stop_daemons();
+
+my ($response_headers, $response_body) = split /\r\n\r\n/, $response, 2;
+
+like($response, qr/HTTP\/1\.1 403 Forbidden/, 'Response returned HTTP 403.');
```
Comment (qiwzhang) [inline src/nginx/t/check_api_target_blocked.t]: also verify the error message "API target is blocked" is returned.

File: repositories.bzl (hunk 1 above)
Comment (kh-chang) [reply repositories.bzl]: The date is below. I can move it up if you think that's better.

Comment (qiwzhang) [reply repositories.bzl]: It is not clear. I though the data below is for the fork SHA. maybe move it up is better.

**Comment by shellscape** [comment]
Will the documentation be updated to reflect these changes? As mentioned in https://github.com/googleapis/googleapis/pull/499, users are left to scour google (with only one useful result) to figure out what this error code means.

**Comment by qiwzhang** [comment]
ESP will send more clear response for the rejected requests.  This PR https://github.com/cloudendpoints/esp/pull/587 will polish the error messages.

**Pull request merged**